### ISSUE TITLE
`jsx-indent`: more bugs

### ISSUE THREAD

**Issue by ljharb** [yannickcr/eslint-plugin-react#898]
```jsx
        <div>
          {[
            <span key="one" />,
            <span key="two" />,
            <span key="three" />,
          ]}
        </div>
```

With it set to "2", i get `Expected indentation of 14 space characters but found 12`.

When I set it to:
```jsx
        <div>
          {[
            <span key="one" />,
              <span key="two" />,
                <span key="three" />,
          ]}
        </div>
```

then `react-jsx-indent` is happy, but `indent` now complains.

**Comment by tleunen** [comment]
Same here. The issue is for any array.

```jsx
[
  'One',
  <span key="two" />,
  'three'
]
```
